lib/train/platforms/detect/helpers/os_windows.rb

```diff
@@ -2,7 +2,12 @@ module Train::Platforms::Detect::Helpers
   module Windows
     def detect_windows
       # try to detect windows, use cmd.exe to also support Microsoft OpenSSH
+      # setting pty to false to execute platform dection code as setting pty to true results in getting unwanted charters in response.
+      # resetting the same once output is gathered.
+
+      set_pty_false if @backend.class.to_s == "Train::Transports::SSH::Connection"
       res = @backend.run_command("cmd.exe /c ver")
```

Comment:
What was the output we were getting here when PTY was enabled?

Reply:
```
"\e[?7l\e[H\e[2J\e[?25l\e[3;1H\e[37m\e[40mBootDevice=\\Device\\HarddiskVolume1\e[4;1HBuildNumber=9600\e[5;1HBuildType=Multiprocessor Free\e[6;1HCaption=Microsoft Windows Server 2012 R2 Datacenter\e[7;1HCodeSet=1252\e[8;1HCountryCode=1\e[9;1HCreationClassName=Win32_OperatingSystem\e[10;1HCSCreationClassName=Win32_ComputerSystem\e[11;1HCSDVersion=\e[12;1HCSName=INS\e[13;1HCurrentTimeZone=0\e[14;1HDataExecutionPrevention_32BitApplications=TRUE\e[15;1HDataExecutionPrevention_Available=TRUE\e[16;1HDataExecutionPrevention_Drivers=TRUE\e[17;1HDataExecutionPrevention_SupportPolicy=3\e[18;1HDebug=FALSE\e[19;1HDescription=\e[20;1HDistributed=FALSE\e[21;1HEncryptionLevel=256\e[22;1HForegroundApplicationBoost=2\e[23;1HFreePhysicalMemory=2563936\e[24;1HFreeSpaceInPagingFiles=1019932\e[24;1H\n\e[JFreeVirtualMemory=3489724\e[24;1H\n\e[JInstallDate=20190624070359.000000+000\e[24;1H\n\e[JLargeSystemCache=\e[24;1H\n\e[JLastBootUpTime=20190624070147.414150+000\e[24;1H\n\e[JLocalDateTime=20190702045319.668000+000\e[24;1H\n\e[JLocale=0409\e[24;1H\n\e[JManufacturer=Microsoft Corporation\e[24;1H\n\e[JMaxNumberOfProcesses=4294967295\e[24;1H\n\e[JMaxProcessMemorySize=137438953344\e[24;1H\n\e[JMUILanguages={\"en-US\"}\e[24;1H\n\e[JName=Microsoft Windows Server 2012 R2 Datacenter|C:\\Windows|\\Device\\Harddisk0\\Pa\e[24;1H\n\e[Jrtition1\e[24;1H\n\e[JNumberOfLicensedUsers=0\e[24;1H\n\e[JNumberOfProcesses=45\e[24;1H\n\e[JNumberOfUsers=3\e[24;1H\n\e[JOperatingSystemSKU=8\e[24;1H\n\e[JOrganization=\e[24;1H\n\e[JOSArchitecture=64-bit\e[24;1H\n\e[JOSLanguage=1033\e[24;1H\n\e[JOSProductSuite=400\e[24;1H\n\e[JOSType=18\e[24;1H\n\e[JOtherTypeDescription=\e[24;1H\n\e[JPAEEnabled=\e[24;1H\n\e[JPlusProductID=\e[24;1H\n\e[JPlusVersionNumber=\e[24;1H\n\e[JPortableOperatingSystem=FALSE\e[24;1H\n\e[JPrimary=TRUE\e[24;1H\n\e[JProductType=3\e[24;1H\n\e[JRegisteredUser=Windows User\e[24;1H\n\e[JSerialNumber=00253-50000-00000-AA442\e[24;1H\n\e[JServicePackMajorVersion=0\e[24;1H\n\e[JServicePackMinorVersion=0\e[24;1H\n\e[JSizeStoredInPagingFiles=1048576\e[24;1H\n\e[JStatus=OK\e[24;1H\n\e[JSuiteMask=400\e[24;1H\n\e[JSystemDevice=\\Device\\HarddiskVolume1\e[24;1H\n\e[JSystemDirectory=C:\\Windows\\system32\e[24;1H\n\e[JSystemDrive=C:\e[24;1H\n\e[JTotalSwapSpaceSize=\e[24;1H\n\e[JTotalVirtualMemorySize=4980328\e[24;1H\n\e[JTotalVisibleMemorySize=3931752\e[24;1H\n\e[JVersion=6.3.9600\e[24;1H\n\e[JWindowsDirectory=C:\\Windows\e[24;1H\n\e[J\n\e[J\n\e[J\n\e[J\e[?12l\e[?25h\e[?7h"
```